drivers/storage/portworx/components_test.go

```diff
@@ -12970,6 +13000,214 @@ func TestServiceTypeAnnotation(t *testing.T) {
+	require.True(t, *cluster.Spec.CSI.InstallSnapshotController)
+	require.NotEmpty(t, cluster.Status.DesiredImages.CSISnapshotController)
+
+	expectedDeployment = testutil.GetExpectedDeployment(t, "csiDeployment_1.0_k8s120.yaml")
+	deployment = &appsv1.Deployment{}
+	err = testutil.Get(driver.k8sClient, deployment, component.CSIApplicationName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedDeployment.Spec, deployment.Spec)
+
+	// TestCase: SnapshotController set to false explicitly
+	reregisterComponents()
+	k8sClient = testutil.FakeK8sClient()
+	driver = portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster = &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1.StorageClusterSpec{
+			Image: "portworx/image:2.6.1",
+			CSI: &corev1.CSISpec{
+				Enabled:                   true,
+				InstallSnapshotController: boolPtr(false),
+			},
+		},
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	require.NotNil(t, cluster.Spec.CSI.InstallSnapshotController)
+	require.False(t, *cluster.Spec.CSI.InstallSnapshotController)
+	require.Empty(t, cluster.Status.DesiredImages.CSISnapshotController)
+
+	expectedDeployment = testutil.GetExpectedDeployment(t, "csiDeployment_1.0_k8s120_without_snapcontroller.yaml")
+	deployment = &appsv1.Deployment{}
+	err = testutil.Get(driver.k8sClient, deployment, component.CSIApplicationName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedDeployment.Spec, deployment.Spec)
+
+	// TestCase: SnapshotController set to true explicitly
+	reregisterComponents()
+	k8sClient = testutil.FakeK8sClient()
+	driver = portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(0))
+
+	cluster = &corev1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1.StorageClusterSpec{
+			Image: "portworx/image:2.6.1",
+			CSI: &corev1.CSISpec{
+				Enabled:                   true,
+				InstallSnapshotController: boolPtr(true),
+			},
+		},
+	}
+	driver.SetDefaultsOnStorageCluster(cluster)
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+	require.NotNil(t, cluster.Spec.CSI.InstallSnapshotController)
+	require.True(t, *cluster.Spec.CSI.InstallSnapshotController)
+	require.NotEmpty(t, cluster.Status.DesiredImages.CSISnapshotController)
+
+	expectedDeployment = testutil.GetExpectedDeployment(t, "csiDeployment_1.0_k8s120.yaml")
+	deployment = &appsv1.Deployment{}
+	err = testutil.Get(driver.k8sClient, deployment, component.CSIApplicationName, cluster.Namespace)
+	require.NoError(t, err)
+	require.Equal(t, expectedDeployment.Spec, deployment.Spec)
+
+	// TestCase: If snapshot controller exists before creating the StorageCluster, don't create the controller
```

Comment:
please have two test cases for snapshot controller exists in same namespace and different namespace